samples/build-duration-logger/index.html

```diff
@@ -139,47 +147,116 @@
         processBuild(build) {
             this.numBuildsInProcess++;
             const buildEventHandlers = this.createBuildEventHandlers(build);
-            const buildEventStream = new EventSource(this.createBuildEventStreamUrl(build.buildId));
-            let buildEventStreamOpen = true;
-
-            buildEventStream.addEventListener('BuildEvent', event => {
-                const buildEventPayload = JSON.parse(event.data);
-                const { eventType } = buildEventPayload.type;
-
-                if (this.allHandledEventTypes.has(eventType)) {
-                    buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
-                }
-            });
-
-            // there isn't an onclose event that we can listen to, but an error event does get triggered
-            // when the end of the stream is reached so use that as a rough proxy for the end of the stream
-            buildEventStream.onerror = event => {
-                if (buildEventStreamOpen) {
+            const buildEventStreamUrl = this.createBuildEventStreamUrl(build.buildId);
+            const buildStream = createServerSideEventStream(buildEventStreamUrl, {
+                oncomplete:  () => {
+                    // there isn't an onclose event that we can listen to, but an error event does get triggered
```

Comment:
Is this comment still valid as we now have the `oncomplete` method?

Reply:
removed